config/v1/types_build.go

```diff
@@ -69,6 +67,18 @@ type ImageLabel struct {
 	Value string `json:"value,omitempty"`
 }
 
+type Proxy struct {
+	// HTTP is the URL of this proxy for HTTP traffic
+	// +optional
+	HTTP string `json:"http,omitempty"`
+	// HTTPSProxy is the URL of this proxy for HTTPS traffic
+	// +optional
+	HTTPS string `json:"https,omitempty"`
+	// NoProxy is the list of the domains for which this proxy should not be used
+	// +optional
+	NoProxy string `json:"noProxy,omitempty"`
+}
```

Comment:
turns out we already have this struct here:
https://github.com/openshift/api/blob/master/build/v1/types.go#L544

you should be able to just use it, I think, though i dunno how we feel about creating cross-api deps.

Reply:
I think we're discouraging cross-api deps - we only depend on core k8s for `config`.

Reply:
Should the strings be pointers so we can distinguish `nil` from empty?

Reply:
here? imho yes.  That's also consistent with the buildconfig api which uses pointers for the strings, though the entire proxy struct field itself is not a pointer:

https://github.com/openshift/api/blob/master/build/v1/types.go#L544
https://github.com/openshift/api/blob/master/build/v1/types.go#L565

Reply:
> here? imho yes. That's also consistent with the buildconfig api which uses pointers for the strings, though the entire proxy struct field itself is not a pointer:

Why would you ever choose to default to "" instead of not having a default value?

Reply:
> same reason we do it everywhere else: preserves user intent.

@bparees you handle that one level higher when specifying `GitProxy` or `DefaultProxy`. 

@deads2k consistency is my main concern here, do we have an example where we have a common structure used in several places in the API, such that we don't copy stuff between dirs? Or is config a good place to start doing so, iow. `build/v1` should be using config's `ProxyConfig`?

Reply:
> @bparees you handle that one level higher when specifying GitProxy or DefaultProxy.

not sufficient, since i can choose to specify 1, 2, or all 3 proxy values (http, https, noproxy), even once i'm specifying a GitProxy or DefaultProxy.